content/ko/docs/reference/access-authn-authz/authentication.md

```diff
@@ -0,0 +1,1360 @@
+---
+# reviewers:
+# - erictune
+# - lavalamp
+# - deads2k
+# - liggitt
+title: 인증
+content_type: concept
+weight: 10
+---
+
+<!-- overview -->
+이 페이지는에서는 인증 개요를 설명한다.
+
+
+<!-- body -->
+## 쿠버네티스의 사용자
+
+모든 쿠버네티스 클러스터는 쿠버네티스가 관리하는 서비스 계정과 일반 사용자
+두 가지 범주의 사용자가 있다.
+
+클러스터-독립적인 서비스는 다음과 같은 방식으로 일반 사용자를 관리한다.
+
+- 개인키(Private key)를 배포하는 관리자
+- Keystone이나 Google 계정과 같은 사용자 저장소
+- 사용자 이름과 비밀번호 목록이 있는 파일
+
+위의 항목에 따르면, _쿠버네티스는 일반 사용자 계정을 나타내는 오브젝트가 없다._
+일반 사용자 계정은 API 호출을 통해 클러스터에 추가될 수 없다.
+
+API 호출을 통해 일반 사용자를 추가할 수는 없지만, 클러스터의 인증서 기관(CA)에 의해 
+서명된 유효한 인증서를 제시하는 모든 사용자는 인증된 것으로 간주된다.
+이 구성에서 쿠버네티스는 인증서의 'subject'의 공통 이름 필드(예: "/CN=bob")에서 
+사용자 이름을 결정한다.
+여기에서 역할 기반 엑세스 제어(RBAC) 하위 시스템은 사용자가 리소스에 대해
+특정 작업을 수행할 수 있는지 여부를 결정한다. 
+자세한 내용은 [인증서 요청](/docs/reference/access-authn-authz/certificate-signing-requests/#normal-user)을 참조
+
+반면, 서비스 계정은 쿠버네티스 API에 의해 관리되는 사용자이다. 
+서비스 계정은 특정 네임스페이스에 바인딩되며, 
+API 서버에 의해 자동으로 생성되거나 API 호출을 통해 수동으로 생성된다. 
+서비스 계정은 `Secrets`로 저장된 일련의 자격 증명과 연결되어 있으며, 
+이는 파드에 마운트되어 클러스터 내부 프로세스가 쿠버네티스 API와 통신할 수 있게 한다.
+
+API 요청은 일반 사용자나 서비스 계정에 바인딩되거나 [익명 요청](#anonymous-requests)으로 처리된다. 
+이는 워크스테이션에서 'kubectl'을 입력하는 인간 사용자부터 노드의 
+'kubelet', 컨트롤 플레인의 멤버에 이르기까지,
+클러스터 내부 또는 외부의 모든 프로세스가 API 서버에 요청을 보낼 때 
+인증해야 하거나 익명 사용자로 처리되어야 함을 의미한다.
+
+## 인증 전략
+
+쿠버네티스는 API 요청을 인증하기 위해 클라이언트 인증서, 베어러 토큰
+또는 인증 프록시를 사용하여 인증 전략을 구현한다. 
+API 서버에 HTTP 요청이 이루어지면 플러그인은 다음과 같은 속성을 요청과 연결하려고 시도한다.
+
+* 사용자 이름(Username): 엔드 유저를 식별하는 문자열이다. 일반적으로 `kube-admin` 또는 `jane@example.com`과 같은 값을 가질 수 있다.
+* 사용자 식별자(UID): 사용자를 식별하는 문자열로, 사용자 이름보다 일관되고 고유하게 설정된다.
+* 그룹(Group): 사용자가 소속된 명명된 논리적인 사용자 집합을 나타내는 문자열의 집합이다. 일반적으로 `system:masters` 또는 `devops-team`과 같은 값을 가질 수 있다.
+* 추가 필드(Extra fields): 인가자(authorizer)가 유용하게 활용할 수 있는 추가 정보를 문자열에서 문자열 리스트로 매핑한 map이다.
+
+모든 값들은 인증 시스템에 대해 불투명하며 [인가자](/ko/docs/reference/access-authn-authz/authorization/)가 해석할 때만 의미를 갖는다.
+
+여러 인증 방법을 동시에 활성화할 수도 있다. 일반적으로 다음과 같은 방법을 사용해야 한다.
+
+- 서비스 계정을 위한 서비스 계정 토큰
+- 사용자 인증을 위한 적어도 하나의 다른 방법
+
+여러 개의 인증 모듈이 활성화된 경우, 
+첫 번째로 요청을 성공적으로 인증하는 모듈이 평가를 중단시킨다. 
+API 서버는 인증자가 실행되는 순서를 보장하지 않는다.
+
+`system:authenticated` 그룹은 모든 인증된 사용자의 그룹 목록에 포함된다.
+
+다른 인증 프로토콜(LDAP, SAML, Kerberos, 대체 x509 스키마 등)과의 통합은 
+[인증 프록시](#authenticating-proxy) 또는
+[인증 웹훅](#webhook-token-authentication)을 사용하여 수행할 수 있다.
+
+### X509 클라이언트 인증서(X509 Client Certs)
+
+클라이언트 인증서를 사용한 인증은 API 서버에 `--client-ca-file=SOMEFILE`옵션을 전달하여 
+활성화된다. 참조된 파일은 API 서버에 제시된 클라이언트 인증서를 검증하는 데 사용할 
+하나 이상의 인증 기관을 포함해야 한다. 클라이언트 인증서가 제시되고 확인된 경우, 
+주체의 공통 이름이 요청의 사용자 이름으로 사용된다. 쿠버네티스 1.4부터는 클라이언트 인증서의 
+조직 필드를 사용하여 사용자의 그룹 소속을 나타낼 수도 있다. 사용자에 대해 여러 그룹 소속을 
+포함하려면 인증서에 여러 조직 필드를 포함하면 된다.
+
+예를 들어, `openssl` 명령 줄 도구를 사용하여 인증서 서명 요청을 생성하는 방법은 다음과 같다.
+
+``` bash
+openssl req -new -key jbeda.pem -out jbeda-csr.pem -subj "/CN=jbeda/O=app1/O=app2"
+```
+
+다음은 "jbeda"라는 사용자 이름을 가진 CSR을 생성하며, "app1"과 "app2"라는 두 개의 그룹에 속하는 예시이다.
```

Comment:
```suggestion
위의 명령은 "jbeda"라는 사용자 이름을 가진 CSR을 생성하며, "app1"과 "app2"라는 두 개의 그룹에 속하게 한다.
```


`This would create a CSR for the username "jbeda", belonging to two groups, "app1" and "app2"`